pkg/auth/authorizer/interfaces.go

```diff
@@ -25,14 +25,8 @@ import (
 // Attributes is an interface used by an Authorizer to get information about a request
 // that is used to make an authorization decision.
 type Attributes interface {
-	// The user string which the request was authenticated as, or empty if
-	// no authentication occurred and the request was allowed to proceed.
-	GetUserName() string
-
-	// The list of group names the authenticated user is a member of. Can be
-	// empty if the authenticated user is not in any groups, or if no
-	// authentication occurred.
-	GetGroups() []string
+	// GetUser returns the user.Info object to authorize
```

Comment:
original interface allowed GetUserName to be empty if no auth occurred... can GetUser return nil in that case?